### ISSUE TITLE
got element 0 of tensors does not require grad and does not have a grad_fn

### ISSUE THREAD

**Issue by smthemex** [ubc-vision/sonic#2]
when run test sonic_inpaint, codes [link](https://github.com/ubc-vision/sonic/blob/main/sonic_inpaint.py#L146)
got this error:
got element 0 of tensors does not require grad and does not have a grad_fn

gradient vanishing ?  test env  is 2.8.0+cu128
what is the torch version should being tested?

**Comment by syeonb** [comment]
Our environment setup uses PyTorch 2.7.1+cu128. We were not able to replicate the error on our end, but just in case, we added an explicit `requires_grad=True` flag for `epsilon_optim_target`. Let us know if the issue persists! 😊

**Comment by smthemex** [comment]
The problem is that a node in my comfyUI env set a global torch.no_grad(). Thanks for the reply, this issue is now resolved.
finally ,after two week  ,fix it ....
sd35 mode example：

<img width="1403" height="820" alt="Image" src="https://github.com/user-attachments/assets/93cab118-1940-4c7b-8e32-52443daf7ff4" />

The solution is to wrap the function with  with torch.inference_mode(False)  and the pipe with with torch.inference_mode(Ture), for example in comfyUI:
```
def function ():
   ...
   with torch.inference_mode(True): # instead  'with torch.no_grad():'
         model_final_output = pipe(
                  prompt="",
                  num_inference_steps=step_nums,

with torch.inference_mode(False)
    def function()
```
